### PR TITLE
feat: remove buy button and landing terminology for uk

### PR DESCRIPTION
**Pull request by JackShort**
We were requested to remove certain items and verbiage regarding UK regulation - you can see here: https://www.notion.so/uniswaplabs/UK-Regulation-Marketing-Updates-0061115831f947a7bd573e86c2f16f3a

The solution involves updating the analytics package to return the origin country and that is used appropriately to display the correct information.

__caching will be implemented in a subsequent pr__
__will also write tests when caching is implemented__

### CODE REVIEW

File: src/analytics/index.tsx [hunk 1]
```diff
@@ -52,3 +52,8 @@ export const sendAnalyticsEvent: typeof sendAnalyticsTraceEvent = (event, proper
     sendAnalyticsTraceEvent(event, properties)
   }
 }
+
+// This is only used for initial page load so we can get the user's country
+export const sendAnalyticsEventIgnoreTracking: typeof sendAnalyticsTraceEvent = (event, properties) => {
+  sendAnalyticsTraceEvent(event, properties)
```
Comment (just-toby) [inline src/analytics/index.tsx:58]: how is this function ignoring tracking? it's just calling the logging function in the same way as `sendAnalyticsEvent` right?

Comment (JackShort) [reply src/analytics/index.tsx:58]: `sendAnalyticsEvent` checks to make sure the analytics ignore key is not set - this does not

File: src/pages/Landing/index.tsx [hunk 2]
```diff
@@ -124,6 +141,12 @@ const TitleText = styled.h1<{ isDarkMode: boolean }>`
   background-clip: text;
   -webkit-background-clip: text;
 
+  ${({ $visible }) =>
```
Comment (just-toby) [inline src/pages/Landing/index.tsx:144]: should there be CSS which hides this component when `$visible` is false?

File: src/pages/Landing/index.tsx [hunk 3]
```diff
@@ -150,9 +173,15 @@ const SubText = styled.div`
   }
 `
 
-const SubTextContainer = styled.div`
+const SubTextContainer = styled.div<{ $visible: boolean }>`
   display: flex;
   justify-content: center;
+
+  ${({ $visible }) =>
```
Comment (just-toby) [inline src/pages/Landing/index.tsx:180]: should there be CSS which hides this component when $visible is false?

File: src/pages/Landing/index.tsx [hunk 4]
```diff
@@ -320,6 +376,34 @@ export default function Landing() {
 
   const location = useLocation()
   const queryParams = parse(location.search, { ignoreQueryPrefix: true })
+
+  const Titles = useMemo(() => {
```
Comment (just-toby) [inline src/pages/Landing/index.tsx:380]: i think this should be. lowercase

File: src/analytics/index.tsx [hunk 5]
```diff
@@ -52,3 +52,8 @@ export const sendAnalyticsEvent: typeof sendAnalyticsTraceEvent = (event, proper
     sendAnalyticsTraceEvent(event, properties)
   }
 }
+
+// This is only used for initial page load so we can get the user's country
```
Comment (just-toby) [inline src/analytics/index.tsx:56]: if you don't want this function to be used in the future (it probably shouldn't), we might want to rename to something like "sendInitializationEvent"

File: src/pages/Landing/index.tsx (hunk 3 above)
Comment (JackShort) [reply src/pages/Landing/index.tsx:180]: nothing gets displayed anyway but i shall just add one since it won't hurt

File: src/pages/Landing/index.tsx [hunk 6]
```diff
@@ -402,8 +479,8 @@ export default function Landing() {
               />
             ))}
           </CardGrid>
-          <CardGrid cols={3}>
-            {MORE_CARDS.map(({ darkIcon, lightIcon, ...card }) => (
+          <CardGrid cols={renderUkSpecificText ? 2 : 3}>
```
Comment (just-toby) [inline src/pages/Landing/index.tsx:482]: can this be less specific? like can we use `extraCards.length` to determine this value? 

it's fine with one parameter but i'm just imagining the cards logic getting more complex

**Review by just-toby — APPROVED** [review]
lgtm

File: src/pages/Landing/index.tsx (hunk 6 above)
Comment (JackShort) [reply src/pages/Landing/index.tsx]: based

File: src/pages/Landing/index.test.tsx [hunk 7]
```diff
@@ -7,7 +9,12 @@ import Landing from '.'
 jest.mock('hooks/useDisableNFTRoutes')
 
 describe('disable nft on landing page', () => {
+  beforeAll(() => {
+    store.dispatch(setOriginCountry('US'))
+  })
+
   it('renders nft information and card', () => {
+    store.dispatch(setOriginCountry('US'))
```
Comment (gnewfield) [inline src/pages/Landing/index.test.tsx:17]: Why do this again if already done in `beforeAll`?

File: src/pages/Landing/index.tsx [hunk 8]
```diff
@@ -303,9 +334,36 @@ export default function Landing() {
   const cardsRef = useRef<HTMLDivElement>(null)
   const selectedWallet = useAppSelector((state) => state.user.selectedWallet)
   const shouldDisableNFTRoutes = useDisableNFTRoutes()
-  const cards = useMemo(
-    () => MAIN_CARDS.filter((card) => !(shouldDisableNFTRoutes && card.to.startsWith('/nft'))),
-    [shouldDisableNFTRoutes]
+  const originCountry = useAppSelector((state: AppState) => state.user.originCountry)
+  const renderUkSpecificText = Boolean(originCountry) && originCountry === 'GB'
```
Comment (gnewfield) [inline src/pages/Landing/index.tsx:338]: Just confirming - will origin country be `GB` or `UK`?

File: src/components/swap/SwapBuyFiatButton.test.tsx [hunk 9]
```diff
@@ -45,6 +47,7 @@ describe('SwapBuyFiatButton.tsx', () => {
   let useOpenModal: jest.Mock<any, any>
 
   beforeAll(() => {
+    store.dispatch(setOriginCountry('US'))
```
Comment (gnewfield) [inline src/components/swap/SwapBuyFiatButton.test.tsx:50]: Can you also add tests for when origin country is `GB` and `null`?

**Review by gnewfield — CHANGES_REQUESTED** [review]

File: src/pages/Landing/index.tsx (hunk 8 above)
Comment (JackShort) [reply src/pages/Landing/index.tsx:338]: GB is the two character ISO code for UK

File: src/components/swap/SwapBuyFiatButton.test.tsx (hunk 9 above)
Comment (JackShort) [reply src/components/swap/SwapBuyFiatButton.test.tsx:50]: I'm adding tests when I implement caching tomorrow because it's just annoying to test with the opacity stuff right now and when caching is implemented it will be easier

File: src/pages/Landing/index.test.tsx (hunk 7 above)
Comment (JackShort) [reply src/pages/Landing/index.test.tsx:17]: oops forgot to delete

File: src/analytics/index.tsx (hunk 1 above)
Comment (justindomingue) [reply src/analytics/index.tsx:58]: ^ probably worth putting this as a comment in code

**Review by tinaszheng — APPROVED** [review]

**Review by gnewfield — APPROVED** [review]
